projects/smoke_thermistor/src/main.c

```diff
@@ -0,0 +1,59 @@
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+// Toggle the following to select which data is logged
+#define READ_DATA_RAW true
+#define READ_DATA_CONVERTED false
+#define NUM_ADC_PINS 6
```

Comment:
With the change from the last comment, instead of hardcoding the number of pins separately from the array of pins, I'd prefer you just use `SIZEOF_ARRAY(s_pin_addresses)` every time instead of having a separate constant.